src/main/java/org/jenkinsci/test/acceptance/plugins/credentials/DomainPage.java

```diff
@@ -28,7 +32,19 @@ public URL getConfigUrl() {
 
     @Override
     public void save() {
-        clickButton("OK");
+        if (driver.getCurrentUrl().contains("configure")) {
```

Comment:
As discussed offline, this is not the Jenkins configuration page but the domain one. Anyway I am improving the detection of such configuration page by checking the whole URL.